From: Richard Kennedy <richard@rsk.demon.co.uk>
Date: Fri, 10 Sep 2010 12:19:43 +0100
Subject: [PATCH] libata: reorder ata_queued_cmd to remove alignment padding on 64 bit builds
X-Git: b34e90429ce8a23546b6b927d4e151df4c113644
References: FATE#311826
Patch-Mainline: 2.6.39

Reorder structure ata_queued_cmd to remove 8 bytes of alignment padding
on 64 bit builds & therefore reduce the size of structure ata_port by
256 bytes.

Overall this will have little impact, other than reducing the amount of
memory that is cleared when allocating ata_ports.

Signed-off-by: Richard Kennedy <richard@rsk.demon.co.uk>
Signed-off-by: Jeff Garzik <jgarzik@redhat.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/libata.h |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/linux/libata.h b/include/linux/libata.h
index ab59f63..a330520 100644
--- a/include/linux/libata.h
+++ b/include/linux/libata.h
@@ -567,13 +567,13 @@ struct ata_queued_cmd {
 	unsigned int		extrabytes;
 	unsigned int		curbytes;
 
-	struct scatterlist	*cursg;
-	unsigned int		cursg_ofs;
-
 	struct scatterlist	sgent;
 
 	struct scatterlist	*sg;
 
+	struct scatterlist	*cursg;
+	unsigned int		cursg_ofs;
+
 	unsigned int		err_mask;
 	struct ata_taskfile	result_tf;
 	ata_qc_cb_t		complete_fn;
-- 
1.6.0.2

